modules/calendar/add.php

```diff
@@ -0,0 +1,548 @@
+                                </script>';
+            echo $alert_close_box;
+            exit();
+          } else {
+            // event is provider event, move onto next event
+          }
+        }
+      }
+    }
+
+    function createHolidayEvent($date, $value, $prov_id) {
+      // $date is event date as a YYYY-MM-DD string
+      // $value is an array consisting of day type string followed by facility id string
+      // $prov_id is id string of a provider
+
+      $event_date = $date;  // event date
+      $event_end_date = "0000-00-00";  // Since, single event
+      // if $value[0] == "Full day"
+      $starttime = "{$GLOBALS['schedule_start']}:00:00";
+      $endtime = "{$GLOBALS['schedule_end']}:00:00";
+      if ($value[0] == "First half") {
+        $starttime = "{$GLOBALS['schedule_start']}:00:00";
+        $endtime = "13:00:00";
+      } elseif ($value[0] == "Second half") {
+        $starttime = "13:00:00";
+        $endtime = "{$GLOBALS['schedule_end']}:00:00";
+      }
+      $provider = $prov_id;
+
+      // Event location specifications
+      $locationspecs = array( "event_location" => "",
+                              "event_street1" => "",
+                              "event_street2" => "",
+                              "event_city" => "",
+                              "event_state" => "",
+                              "event_postal" => "" );
+      $locationspec = serialize($locationspecs);
+
+      // Recurring event specifications
+      $recurrspec = array( "event_repeat_freq" => "0",
+                           "event_repeat_freq_type" => "0",
+                           "event_repeat_on_num" => "1",
+                           "event_repeat_on_day" => "0",
+                           "event_repeat_on_freq" => "0",
+                           "exdate" => "" );
+
+      $args = array();
+        // specify values needed for the INSERT into table - libreehr_postcalendar_events - columns
+        $args['form_category'] = "6";  // Holiday
+        $args['new_multiple_value'] = "";
+        $args['form_provider'] = $provider;  // selected provider
+        $args['form_title'] = "Holidays";
+        $args['form_comments'] = "";
+        $args['event_date'] = $event_date;
+        $args['form_enddate'] = $event_end_date;
+        $args['duration'] = 0;  // no duration in provider events
+        $args['form_repeat'] = "0";  // Repeats box not checked
+        $args['recurrspec'] = $recurrspec;
+        $args['starttime'] = $starttime;
+        $args['endtime'] = $endtime;
+        $args['form_allday'] = 0;
+        $args['form_apptstatus'] = "-";  // None
+        $args['form_prefcat'] = "0";  // None
+        $args['locationspec'] = $locationspec;
+        // Taking facility & billing facility as selected facility of holiday
+        $args['facility'] = $value[1];
+        $args['billing_facility'] = $value[1];
```

Comment:
Billing facility in non provider events is important and could be different from the facility. It does not matter with provider events just want you to know about it .